src/Modules/Legacy/String/PrintStringIntoString.cc

```diff
@@ -125,34 +120,33 @@ PrintStringIntoString::execute()
 
           std::string fstr = format.substr(i,j-i+1);
 
+          if ((format[j] == 's')||(format[j] == 'S')||(format[j] == 'c')||(format[j] == 'C'))
           {
-            str = "";
-            if (lastport == false)
             {
-              if (inputport == stringsH.size())
-              {
-                lastport = true;
-              }
-              else
+              str = "";
+              if (!lastport)
               {
-                if (stringsH.size() == static_cast<size_t>(inputport))
+                if (inputport == stringsH.size())
                 {
                   lastport = true;
                 }
                 else
                 {
-                  currentstring = stringsH[inputport]; inputport++;
-                  if (currentstring)
+                  if (stringsH.size() == static_cast<size_t>(inputport))
+                  {
+                    lastport = true;
+                  }
+                  else
                   {
-                    str = currentstring->value();
+                    currentstring = stringsH[inputport]; inputport++;
```

Comment:
How about:
`currentstring = stringsH[inputport++];`

Reply:
Done. Now of course, Github needs you to re-approve.....grr